pabi_procurement/models/purchase_web_interface.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, api, _
+from openerp.exceptions import Warning as UserError
+import xmlrpclib
+import base64
+import os
+import inspect
+
+
+class PurchaseWebInterface(models.Model):
+    _name = "purchase.web.interface"
+    _description = "Purchase Web Interface"
+
+    @api.model
+    def encode_base64(self, filename):
+        current_path = inspect.getfile(inspect.currentframe())
+        directory_path = os.path.dirname(os.path.abspath(current_path))
+        path = os.path.expanduser(directory_path+'/../data/'+filename)
+        with open(path) as f:
+            encoded = base64.b64encode(f.read())
+            return encoded
+
+    @api.model
+    def check_pdf_extension(self, filename):
+        if '.pdf' not in filename:
+            filename += '.pdf'
+        return filename
+
+    @api.model
+    def send_pd_test(self):
+        ConfParam = self.env['ir.config_parameter']
+        url = ConfParam.get_param('pabiweb_url')
+        username = ConfParam.get_param('pabiweb_username')
+        password = ConfParam.get_param('pabiweb_password')
+        connect_string = "http://"+username+":"+password+"@"+url
```

Comment:
In Python best practice, it should be
connect_string = "http://%s:%s@%s" % (username, password, url)